pkg/pipeline-selector/pipeline_selector.go

```diff
@@ -0,0 +1,162 @@
+	selectionParameters, err := getPipelineSelectionParametersForComponent(component)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for i := range selectors {
+		if buildPipelineRef, buildPipelineAdditionalParams := findMatchingPipeline(selectionParameters, &selectors[i]); buildPipelineRef != nil {
+			return buildPipelineRef, buildPipelineAdditionalParams, nil
+		}
+	}
+	return nil, nil, nil
+}
+
+// getPipelineSelectionParametersForComponent returns build parameters of the given component
+// to use when matching to the build pipelines selection logic.
+func getPipelineSelectionParametersForComponent(component *appstudiov1alpha1.Component) (*buildappstudiov1alpha1.WhenCondition, error) {
+	parameters := &buildappstudiov1alpha1.WhenCondition{}
+
+	parameters.ComponentName = component.GetName()
+	parameters.Annotations = component.GetAnnotations()
+	parameters.Labels = component.GetLabels()
+
+	devfileData, err := devfile.ParseDevfileModel(component.Status.Devfile)
+	if err != nil {
+		return nil, err
+	}
+
+	devfileMetadata := devfileData.GetMetadata()
+
+	parameters.Language = devfileMetadata.Language
+	parameters.ProjectType = devfileMetadata.ProjectType
+
+	var dockerfileRequired bool
+	if dockerfile, err := devfile.SearchForDockerfile([]byte(component.Status.Devfile)); err == nil && dockerfile != nil {
+		dockerfileRequired = true
+		parameters.DockerfileRequired = &dockerfileRequired
+	} else {
+		dockerfileRequired = false
+		parameters.DockerfileRequired = &dockerfileRequired
+	}
+
+	return parameters, nil
+}
+
+// findMatchingPipeline evaluates given selectors chain against component parameters.
+// The first match is returned.
+func findMatchingPipeline(selectionParameters *buildappstudiov1alpha1.WhenCondition, selectors *buildappstudiov1alpha1.BuildPipelineSelector) (*tektonapi.PipelineRef, []tektonapi.Param) {
+	for _, pipelineSelector := range selectors.Spec.Selectors {
+		if pipelineConditionsMatchComponentParameters(&pipelineSelector.WhenConditions, selectionParameters) {
+			var pipelineParams []tektonapi.Param
+			for _, param := range pipelineSelector.PipelineParams {
+				pipelineParams = append(pipelineParams, tektonapi.Param{
+					Name:  param.Name,
+					Value: *tektonapi.NewArrayOrString(param.Value),
+				})
+			}
+			return &pipelineSelector.PipelineRef, pipelineParams
+		}
+	}
+	return nil, nil
+}
+
+// pipelineConditionsMatchComponentParameters evaluates given pipeline selector against component parameters.
+// In other words, checks if given pipeline can build the component (according to what the pipeline conditions say).
+func pipelineConditionsMatchComponentParameters(pipeline, component *buildappstudiov1alpha1.WhenCondition) bool {
+	if pipeline.Language != "" && !pipelineMatchesComponentCondition(pipeline.Language, component.Language) {
```

Comment:
switch {
case ...
}

is generally the preferred form vs. if's for what you have here, but I'll still apply the approve label and let you decide on your own if you want to change